sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/transfer/TransferIdInfoLiveData.java

```diff
@@ -241,20 +285,44 @@ private void setLastWorkInfoState(WorkInfo.State state) {
      */
     private void mapErrorFromTransferClient() {
         if (this.inputTransferIdOrError.isError()) {
-            this.transferInfoLiveData.setValue(TransferInfo.createFailed(this.inputTransferIdOrError.getId(),
-                this.inputTransferIdOrError.getErrorMessage()));
-            this.setLastWorkInfoState(WorkInfo.State.FAILED);
-            return;
+            if (this.inputTransferIdOrError.getOperation() == TransferIdOrError.Operation.RESUME) {
+                if (this.inputTransferIdOrError.isNotFoundError()) {
+                    // Follow the same convention as WorkManager, i.e. if an Work is
+                    // unknown then emit null. Here application provided transfer id was not
+                    // identifying a transfer record.
+                    this.transferInfoLiveData.setValue(null);
+                    this.setLastWorkInfoState(WorkInfo.State.FAILED);
+                    return;
+                } else if (this.inputTransferIdOrError.isTerminatedError()) {
+                    TransferIdOrError.TransferInTerminatedStateError tError = this.inputTransferIdOrError.getError();
+                    if (tError.isCompleted()) {
+                        this.transferInfoLiveData
+                            .setValue(TransferInfo.createCompleted(this.inputTransferIdOrError.getId()));
+                        this.setLastWorkInfoState(WorkInfo.State.SUCCEEDED);
+                        return;
+                    }
+                }
+                this.transferInfoLiveData.setValue(TransferInfo.createFailed(this.inputTransferIdOrError.getId(),
+                    this.inputTransferIdOrError.getErrorMessage()));
+                this.setLastWorkInfoState(WorkInfo.State.FAILED);
+                return;
+            } else {
+                // TransferIdOrError.Operation.UPLOAD_DOWNLOAD
```

Comment:
Unclear what this comment is for

Reply:
I mean, this code path hit for upload or download not for resume. I will make it more clear in the doc instead of just putting enum.